api/src/main/kotlin/com/gmtkgamejam/search/Opensearch.kt

```diff
@@ -0,0 +1,90 @@
+package com.gmtkgamejam.search
+
+import com.fasterxml.jackson.databind.node.ObjectNode
+import com.gmtkgamejam.models.posts.SearchItem
+import org.apache.hc.core5.http.HttpHost
+import org.opensearch.client.json.jackson.JacksonJsonpMapper
+import org.opensearch.client.opensearch.OpenSearchClient
+import org.opensearch.client.opensearch._types.SortOptions
+import org.opensearch.client.opensearch._types.query_dsl.Query
+import org.opensearch.client.opensearch.core.DeleteRequest
+import org.opensearch.client.opensearch.core.IndexRequest
+import org.opensearch.client.opensearch.core.SearchRequest
+import org.opensearch.client.opensearch.core.UpdateRequest
+import org.opensearch.client.transport.OpenSearchTransport
+import org.opensearch.client.transport.httpclient5.ApacheHttpClient5TransportBuilder
+
+/**
+ * Singleton access point to the Opensearch cluster
+ */
+object OpenSearch {
+
+    private val client: OpenSearchClient
+
+    init {
+        val transport: OpenSearchTransport = ApacheHttpClient5TransportBuilder
+            .builder(HttpHost("http", "localhost", 9200))
+            .setMapper(JacksonJsonpMapper())
+            .build()
+
+        client = OpenSearchClient(transport)
+    }
+
+    /**
+     * Perform a search request against OpenSearch and return the document IDs in the result
+     */
+    fun search(query: Query, sortOptions: List<SortOptions>): List<String> {
+        val searchRequest = SearchRequest.Builder()
+            .index("posts")
+            .query(query)
+            .size(32)
+            .sort(sortOptions)
+            .build()
+
+        return client.search(searchRequest, ObjectNode::class.java).hits().hits()
+            .map { it.source()!!.get("id").asText() }
+            .toList()
```

Comment:
It would have done, but the `source` is the indexed source data, so it should be impossible to trigger this.

_Should_

I'll commit your change now, thanks again @awildbrysen :D